esp-hal-common/src/rtc_cntl/sleep/esp32c6.rs

```diff
@@ -0,0 +1,970 @@
+                let mut pd_flags = PowerDownFlags(0);
+
+                pd_flags.set_pd_lp_periph(true);
+                pd_flags.set_pd_cpu(true);
+                pd_flags.set_pd_xtal32k(true);
+                pd_flags.set_pd_rc32k(true);
+                pd_flags.set_pd_rc_fast(true);
+                pd_flags.set_pd_xtal(true);
+                pd_flags.set_pd_top(true);
+                pd_flags.set_pd_modem(true);
+                pd_flags.set_pd_vddsdio(true);
+
+                pd_flags
+            },
+        }
+    }
+}
+
+unsafe fn pmu<'a>() -> &'a esp32c6::pmu::RegisterBlock {
+    &*esp32c6::PMU::ptr()
+}
+
+unsafe fn lp_aon<'a>() -> &'a esp32c6::lp_aon::RegisterBlock {
+    &*esp32c6::LP_AON::ptr()
+}
+
+bitfield::bitfield! {
+    #[derive(Clone, Copy)]
+    /// Power domains to be powered down during sleep
+    pub struct PowerDownFlags(u32);
+
+    pub u32, pd_top      , set_pd_top      : 0;
+    pub u32, pd_vddsdio  , set_pd_vddsdio  : 1;
+    pub u32, pd_modem    , set_pd_modem    : 2;
+    pub u32, pd_hp_periph, set_pd_hp_periph: 3;
+    pub u32, pd_cpu      , set_pd_cpu      : 4;
+    pub u32, pd_hp_aon   , set_pd_hp_aon   : 5;
+    pub u32, pd_mem_g0   , set_pd_mem_g0   : 6;
+    pub u32, pd_mem_g1   , set_pd_mem_g1   : 7;
+    pub u32, pd_mem_g2   , set_pd_mem_g2   : 8;
+    pub u32, pd_mem_g3   , set_pd_mem_g3   : 9;
+    pub u32, pd_xtal     , set_pd_xtal     : 10;
+    pub u32, pd_rc_fast  , set_pd_rc_fast  : 11;
+    pub u32, pd_xtal32k  , set_pd_xtal32k  : 12;
+    pub u32, pd_rc32k    , set_pd_rc32k    : 13;
+    pub u32, pd_lp_periph, set_pd_lp_periph: 14;
+}
+
+impl PowerDownFlags {
+    pub fn pd_mem(self) -> bool {
+        self.pd_mem_g0() && self.pd_mem_g1() && self.pd_mem_g2() && self.pd_mem_g3()
+    }
+
+    pub fn set_pd_mem(&mut self, value: bool) {
+        self.set_pd_mem_g0(value);
+        self.set_pd_mem_g1(value);
+        self.set_pd_mem_g2(value);
+        self.set_pd_mem_g3(value);
+    }
+}
+
+struct MachineConstants;
+impl MachineConstants {
+    const LP_MIN_SLP_TIME_US: u32 = 450;
+    const LP_WAKEUP_WAIT_CYCLE: u32 = 4;
+    const LP_ANALOG_WAIT_TIME_US: u32 = 154;
+    const LP_XTAL_WAIT_STABLE_TIME_US: u32 = 250;
+    const LP_CLK_SWITCH_CYCLE: u32 = 1;
+    const LP_CLK_POWER_ON_WAIT_CYCLE: u32 = 1;
+    const LP_POWER_SUPPLY_WAIT_TIME_US: u32 = 2;
+    const LP_POWER_UP_WAIT_TIME_US: u32 = 2;
+
+    const HP_MIN_SLP_TIME_US: u32 = 450;
+    const HP_CLOCK_DOMAIN_SYNC_TIME_US: u32 = 150;
+    const HP_SYSTEM_DFS_UP_WORK_TIME_US: u32 = 124;
+    const HP_ANALOG_WAIT_TIME_US: u32 = 154;
+    const HP_POWER_SUPPLY_WAIT_TIME_US: u32 = 2;
+    const HP_POWER_UP_WAIT_TIME_US: u32 = 2;
+    const HP_REGDMA_S2M_WORK_TIME_US: u32 = 172;
+    const HP_REGDMA_S2A_WORK_TIME_US: u32 = 480;
+    const HP_REGDMA_M2A_WORK_TIME_US: u32 = 278;
+    // const HP_REGDMA_A2S_WORK_TIME_US: u32 = 382;
+    const HP_REGDMA_RF_ON_WORK_TIME_US: u32 = 70;
+    // const HP_REGDMA_RF_OFF_WORK_TIME_US: u32 = 23;
```

Comment:
these are constants defined in esp-idf but unused. I kept them for completeness, but commented to avoid the warnings. I suspect they will serve some function later.

Reply:
Yeah, probably if they aren't supposed to be commented out there would be compile errors 🙃 Friday evening is not the best time for reviewing such PR, haha :D